update.js

```diff
@@ -0,0 +1,43 @@
+import { success, failure } from "./libs/response-lib";
+import * as dynamoDbLib from "./libs/dynamodb-lib";
+import User from './model/user';
+import HttpStatus from 'http-status-codes';
+import { userValidationCriteria, UUID_V1 } from './validators/user';
+import validate from 'validate.js';
+
+/**
+ * Update API Lambda for User model.
+ */
+export async function main(event, context) {
+  try {
+    if (!('pathParameters' in event)) {
+      return failure(HttpStatus.UNPROCESSABLE_ENTITY, {error : "ID Path Parameter required."});
+    }
+    const id = event.pathParameters.id;
+    const data = JSON.parse(event.body);
+    const errors = validate(data, userValidationCriteria);
+
+    if (!id) {
+      return failure(HttpStatus.UNPROCESSABLE_ENTITY, {error : "ID Path Parameter required."});
+    } else if (!id.match(UUID_V1)) {
+      throw {message : "Required UUID V1 format."};
+    } else if (errors) {
+      return failure(HttpStatus.BAD_REQUEST, {errors : errors});
+    }
+
+    let user = new User(dynamoDbLib);
+    try {
+      const exists = await user.read(id);
+      if (!exists) {
+        return failure(HttpStatus.NOT_FOUND, {});
+      }
+      const result = await user.update(id, data);
+      return success(HttpStatus.OK, result);
+    } catch (e) {
+      return failure(HttpStatus.INTERNAL_SERVER_ERROR, {});
+    }
+  } catch (e) {
+    console.info("Invalid request, body: " + event.body + " pathParameters: " + JSON.stringify(event.pathParameters));
+    return failure(HttpStatus.UNPROCESSABLE_ENTITY, {});
```

Comment:
Can you explain the decision behind making `422` a default error?

Reply:
Its not a default, this requires an exception to be thrown to be returned.

Reply:
I understand it needs error to be thrown but this is a "cover-all" error handler. All kinds of errors could happen here. You chose to return 422 from it instead of the usual 500 and I wanted to know what was the reason for this.

Reply:
The reason was simply that I was expecting the catch to be either of my throws in this actual segment of code.  Given that the user model does throw an error, this could lead to a cover all capturing a real 500 and returning the 422, so I see the deficiency here.